src/components/document-content.tsx

```diff
@@ -159,29 +173,54 @@ export class DocumentContentComponent extends BaseComponent<IProps, IState> {
   }
 
   private handleMoveTileDrop = (e: React.DragEvent<HTMLDivElement>) => {
+    const kSideDropThreshold = 20;
     const { content } = this.props;
     if (!content) return;
     const dragTileId = e.dataTransfer.getData(kDragTileId);
     const srcRowId = content.findRowContainingTile(dragTileId);
     if (!srcRowId) return;
     const srcRowIndex = content.rowOrder.findIndex(rowId => rowId === srcRowId);
-    const dropRowIndex = this.getDropRowIndex(e);
+    const dropRowInfo  = this.getDropRowInfo(e);
+    const { rowInsertIndex, rowDropIndex, dropOffsetLeft,
+            dropOffsetTop, dropOffsetRight, dropOffsetBottom } = dropRowInfo;
+    if ((rowDropIndex != null) &&
+        (dropOffsetLeft != null) &&
+        (dropOffsetLeft < kSideDropThreshold) &&
+        (dropOffsetLeft < dropOffsetRight!)) {
+      content.moveTileToRow(dragTileId, rowDropIndex, 0);
+      return;
+    }
+    if ((rowDropIndex != null) &&
+        (dropOffsetRight != null) &&
+        (dropOffsetRight < kSideDropThreshold) &&
+        (dropOffsetRight <= dropOffsetLeft!)) {
+      content.moveTileToRow(dragTileId, rowDropIndex);
+      return;
+    }
```

Comment:
This is very readable! Doesn't work for moving tiles in the middle, but that's probably fine for now